core/src/main/java/org/apache/iceberg/TableMetadata.java

```diff
@@ -307,6 +311,13 @@ public String toString() {
     Preconditions.checkArgument(
         metadataFileLocation == null || changes.isEmpty(),
         "Cannot create TableMetadata with a metadata location and changes");
+    Preconditions.checkArgument(
+        !rowLineageEnabled || formatVersion == 3,
+        "Row lineage is only supported in v3 (current version v%s)",
+        formatVersion);
+    Preconditions.checkArgument(
+        !rowLineageEnabled || nextRowId >= 0,
```

Comment:
Theoretically, `next-row-id` can start from any integer, including negative ones. I am thinking may be in the implementation we could make it starts from 0 to simplify the implementation and make it more intuitive.